test/e2e/plugins_test.go

```diff
@@ -0,0 +1,210 @@
+		assert.Assert(t, knPluginPath != "")
+
+		knPluginPath2, err = test.createPlugin(t, "kn-hello2e2e", TestPluginCode, knPluginsDir2)
+		assert.Assert(t, err == nil)
+		assert.Assert(t, knPluginPath2 != "")
+	}
+
+	teardown := func(t *testing.T) {
+		err = os.RemoveAll(knConfigDir)
+		assert.Assert(t, err == nil)
+	}
+
+	t.Run("when kn config is empty", func(t *testing.T) {
+		t.Run("when using --plugin-dir", func(t *testing.T) {
+			t.Run("when --plugins-dir has a plugin", func(t *testing.T) {
+				t.Run("when --lookup-plugins is false", func(t *testing.T) {
+					lookupPlugins = false
+
+					setup(t)
+					defer teardown(t)
+
+					knFlags := []string{fmt.Sprintf("--plugins-dir=%s", knPluginsDir), "--lookup-plugins=false"}
+					assert.Assert(t, lookupPlugins == false)
+
+					t.Run("list plugin in --plugins-dir", func(t *testing.T) {
+						test.listPlugin(t, knFlags, []string{knPluginPath}, []string{})
+					})
+
+					t.Run("execute plugin in --plugins-dir", func(t *testing.T) {
+						test.runPlugin(t, knFlags, "helloe2e", []string{"e2e", "test"}, []string{"Hello Knative, I'm a Kn plugin", "I recieved arguments: e2e"})
+					})
+
+					t.Run("does not list any other plugin in $PATH", func(t *testing.T) {
+						test.listPlugin(t, knFlags, []string{knPluginPath}, []string{knPluginPath2})
+					})
+				})
+
+				t.Run("with --lookup-plugins is true", func(t *testing.T) {
+					lookupPlugins = true
+
+					setup(t)
+					defer teardown(t)
+
+					knFlags := []string{fmt.Sprintf("--plugins-dir=%s", knPluginsDir), "--lookup-plugins=true"}
+					assert.Assert(t, lookupPlugins == true)
+
+					t.Run("list plugin in --plugins-dir", func(t *testing.T) {
+						test.listPlugin(t, knFlags, []string{knPluginPath}, []string{knPluginPath2})
+					})
+
+					t.Run("execute plugin in --plugins-dir", func(t *testing.T) {
+						test.runPlugin(t, knFlags, "helloe2e", []string{}, []string{"Hello Knative, I'm a Kn plugin"})
+					})
+
+					t.Run("when other plugins are in $PATH", func(t *testing.T) {
+						var oldPath string
+
+						setupPath := func(t *testing.T) {
+							oldPath = os.Getenv("PATH")
+							err := os.Setenv("PATH", fmt.Sprintf("%s:%s", oldPath, knPluginsDir2))
+							assert.Assert(t, err == nil)
+						}
+
+						tearDownPath := func(t *testing.T) {
+							err = os.Setenv("PATH", oldPath)
+							assert.Assert(t, err == nil)
+						}
+
+						t.Run("list plugin in $PATH", func(t *testing.T) {
+							setupPath(t)
+							defer tearDownPath(t)
+							test.listPlugin(t, knFlags, []string{knPluginPath, knPluginPath2}, []string{})
+						})
+
+						t.Run("execute plugin in $PATH", func(t *testing.T) {
+							setupPath(t)
+							defer tearDownPath(t)
+							test.runPlugin(t, knFlags, "hello2e2e", []string{}, []string{"Hello Knative, I'm a Kn plugin"})
+						})
+					})
+				})
+			})
+		})
+	})
+}
+
+// Private
+
+func (test *e2eTest) createConfig(t *testing.T, configName, configCode, configPath string) (string, error) {
```

Comment:
```suggestion
func (test *e2eTest) createConfig(t *testing.T, configFilename, configData, configDir string) (string, error) {
```
(symbol renames)

actually should be ordered as Dir, Filename, Data


```suggestion
func (test *e2eTest) createConfig(t *testing.T, configDir, configFilename, configData string) (string, error) {
```